src/core/Directus/Services/TablesService.php

```diff
@@ -735,6 +740,19 @@ public function getTableObject($tableName)
         return SchemaService::getCollection($tableName);
     }
 
+    public function hasPrimaryField($fields){
+        $result = false;
+
+        foreach($fields as $field){
+            if($field[primary_key] == true){
```

Comment:
@danielfrentz it looks great, but you have to make sure the array key is a string instead of a constant.

`$field['primary_key']` instead of `$field[primary_key]`

Reply:
And maybe use a strict equal check? `===` vs `==` 🙂